src/diffusers/pipelines/free_noise_utils.py

```diff
@@ -30,6 +34,100 @@
 logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
 
 
+class SplitInferenceModule(nn.Module):
+    def __init__(
+        self,
+        module: nn.Module,
+        split_size: int = 1,
+        split_dim: int = 0,
+        input_kwargs_to_split: List[str] = ["hidden_states"],
+    ) -> None:
+        super().__init__()
+
```

Comment:
Would also add a docstring here to explain the init arguments. Maybe the workflow example in forward can be moved up here too?

Reply:
Sounds good!